libraries/tracker-core/src/core.ts

```diff
@@ -429,7 +429,7 @@ export function trackerCore(configuration: CoreConfiguration = {}): TrackerCore
      * @param key - Field name
      * @param value - Field value
      */
-    function addPayloadPair(key: string, value: string | number): void {
+    function addPayloadPair(key: string, value: unknown): void {
```

Comment:
Was this added intentionally in this PR? Doesn't seem related to the dimension issue.

Reply:
Yeap, it came up as the typings complained in the `resolution` result from the dimension values function return type. This showcases this inconsistency between  `addPayloadPair` and `payloadBuilder.add`. Added this in this PR as a small fix/housekeeping.

WDYT ?

Reply:
Ah, I see! That makes sense.